convert2rhel/pkghandler.py

```diff
@@ -358,7 +395,8 @@ def _get_installed_pkg_objects_yum(name=None, version=None, release=None, arch=N
 
         return yum_base.rpmdb.returnPackages(patterns=[pattern])
 
-    return yum_base.rpmdb.returnPackages()
+    installed_packages = yum_base.rpmdb.returnPackages()
```

Comment:
Shouldn't we close and delete the yum base here to have the rpm database properly closed as on the other places? The same question applies to the other use of the yum base in the `_get_original_releasever()` function.
```suggestion
    base.close()
    del base

    installed_packages = yum_base.rpmdb.returnPackages()
```

Reply:
We are not trying to forcefully close the db on every possible case as it can come down to performance issues, as the closing/deleting the object (which will do the same as close), will clean up the rpmdb, which can slow us down a lot if we start closing it after every call. 

We are trying to only close the rpmdb in cases where we notice that yum/rpm is leaving locks behind and causing weird and strange messages to be seen by the user.

Reply:
Those functions are fine because they are running within the same process as convert2rhel. We only had to forcefully close the rpmdb in cases where we are running the functions as a child process. 

I can't see a good place to share this information in the code, so I think I will make a note to myself to append that to our wiki, as I really plan to make a very comprehensive document about working with yum/dnf/rpm here.